### PR TITLE
swift 5.7.2

### PR DESCRIPTION
**Pull request by Bo98**
Created by `brew update-swift`. 🚀

---

Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by cho-m** [comment]
Going to cancel run. Intel Big Sur has been running for 1d 17h while previous v5.7 took 8h. Also, Monterey and Linux already failed.

**Comment by Bo98** [comment]
Fun one:

```
Assertion failed: (match.isViable()), function findBestWitness, file TypeCheckProtocol.cpp, line 1472.
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the project and the crash backtrace.
Stack dump:
0.	Program arguments: /usr/local/Cellar/swift/5.7.1/Swift-5.7.xctoolchain/usr/bin/swift-frontend -compile-module-from-interface -sdk /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -prebuilt-module-cache-path /usr/local/Cellar/swift/5.7.1/Swift-5.7.xctoolchain/usr/lib/swift/macosx/prebuilt-modules/13.0 -module-name Foundation -o /usr/local/Cellar/swift/5.7.1/Swift-5.7.xctoolchain/usr/lib/swift/macosx/prebuilt-modules/13.0/Foundation.swiftmodule/x86_64-apple-ios-macabi.swiftmodule /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks/Foundation.framework/Modules/Foundation.swiftmodule/x86_64-apple-ios-macabi.swiftinterface -suppress-warnings -Fsystem /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/iOSSupport/System/Library/Frameworks -serialize-parseable-module-interface-dependency-hashes -bad-file-descriptor-retry-count 30
1.	Swift version 5.7.1 (swift-5.7.1-RELEASE)
2.	Compiling with the current language version
3.	While evaluating request TypeCheckSourceFileRequest(source_file "/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks/Foundation.framework/Modules/Foundation.swiftmodule/x86_64-apple-ios-macabi.swiftinterface")
4.	While type-checking extension of Measurement.AttributedStyle (at /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks/Foundation.framework/Modules/Foundation.swiftmodule/x86_64-apple-ios-macabi.swiftinterface:4620:1)
5.	While type-checking 'ByteCount' (at /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks/Foundation.framework/Modules/Foundation.swiftmodule/x86_64-apple-ios-macabi.swiftinterface:4621:10)
6.	While evaluating request ResolveImplicitMemberRequest(Foundation.(file).AttributedStyle extension.ByteCount@/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks/Foundation.framework/Modules/Foundation.swiftmodule/x86_64-apple-ios-macabi.swiftinterface:4621:17, resolve CodingKeys)
7.	While type-checking protocol conformance to 'Decodable' (in module 'Swift') for type 'Measurement<UnitType>.AttributedStyle.ByteCount' (declared at [/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/System/Library/Frameworks/Foundation.framework/Modules/Foundation.swiftmodule/x86_64-apple-ios-macabi.swiftinterface:4621:10 - line:4656:3]
```

**Comment by mrstegeman** [comment]
Swift 5.7.2 is now available.

**Comment by Bo98** [comment]
So the crash still happens even in 5.8. It seems that Apple have kept their fix for the issue closed source, or somehow build their Swift to not have the issue while making their official swift.org builds still do.

I've disabled the macOS 13 SDK precompile as a result. This means `brew test swift` will be close to timeout on macOS 13 and maybe macOS 12. Will see how it goes.

I've disabled 10.15 SDK precompile since the Swift minimum macOS has been Big Sur for some time, and I think that might have caused the macOS 11 hang.

I've included an experimental patch on Linux to hopefully fix the occasional build failure due to a race condition. This will be submitted upstream if it works.

File: Formula/swift.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ class Swift < Formula
   # This is mostly community sourced, so may be not necessarily be accurate.
   depends_on xcode: ["13.0", :build]
 
-  depends_on "python@3.10"
+  depends_on "python@3.11"
 
   # HACK: this should not be a test dependency but is due to a limitation with fails_with
   uses_from_macos "llvm" => [:build, :test]
```
Comment (cho-m) [inline Formula/swift.rb:40]: Should be able to remove hack, but probably not worth doing it right now when we are trying to finish this to ship `icu4c` update.

**Comment by cho-m** [comment]
Linux:
```
FAILED: libswiftDispatch.so src/swift/CMakeFiles/swiftDispatch.dir/Block.swift.o src/swift/CMakeFiles/swiftDispatch.dir/Data.swift.o src/swift/CMakeFiles/swiftDispatch.dir/Dispatch.swift.o src/swift/CMakeFiles/swiftDispatch.dir/IO.swift.o src/swift/CMakeFiles/swiftDispatch.dir/Private.swift.o src/swift/CMakeFiles/swiftDispatch.dir/Queue.swift.o src/swift/CMakeFiles/swiftDispatch.dir/Source.swift.o src/swift/CMakeFiles/swiftDispatch.dir/Time.swift.o src/swift/CMakeFiles/swiftDispatch.dir/Wrapper.swift.o src/swift/swift/Dispatch.swiftmodule
: && /tmp/swift-20221224-13814-lq7zjh/build/swift-linux-x86_64/bin/swiftc -output-file-map src/swift/CMakeFiles/swiftDispatch.dir/Release/output-file-map.json -incremental -j 8 -emit-library -o libswiftDispatch.so -module-name Dispatch -module-link-name swiftDispatch -emit-module -emit-module-path src/swift/swift/Dispatch.swiftmodule -emit-dependencies -DswiftDispatch_EXPORTS -module-cache-path "/tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64/module-cache" -O -Xcc -fblocks -Xcc -fmodule-map-file=/tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64/dispatch/module.modulemap -Xcc -I/tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch -Xcc -I/tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/shims -I /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/BlocksRuntime -I /tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64 -I /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch -I /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src -I /tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64/src /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Block.swift /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Data.swift /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Dispatch.swift /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/IO.swift /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Private.swift /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Queue.swift /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Source.swift /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Time.swift /tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Wrapper.swift -no-toolchain-stdlib-rpath -Xlinker -soname -Xlinker libswiftDispatch.so  -L /tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64/src/swift  -L /tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64  -L /tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64   -L /usr/lib/gcc/x86_64-linux-gnu/11   -L /tmp/swift-20221224-13814-lq7zjh/build/llvm-linux-x86_64/lib -Xlinker -rpath -Xlinker /tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64:  src/swift/libDispatchStubs.a  libdispatch.so  libBlocksRuntime.so  -lstdc++  -lm  -lgcc_s  -lgcc  -lc  -lgcc_s  -lgcc && :
/tmp/swift-20221224-13814-lq7zjh/build/libdispatch-linux-x86_64/dispatch/module.modulemap:14:18: error: umbrella header 'dispatch.h' not found
        umbrella header "dispatch.h"
                        ^
/tmp/swift-20221224-13814-lq7zjh/swift-corelibs-libdispatch/src/swift/Block.swift:13:8: error: could not build C module 'CDispatch'
import CDispatch
       ^
```

**Comment by Bo98** [comment]
That one's my fault - will take a look tomorrow if I can. If not, Boxing Day.

**Comment by cho-m** [comment]
```
==> /home/linuxbrew/.linuxbrew/Cellar/swift/5.7.2/bin/swift -module-cache-path /tmp/swift-test-20230104-125986-17lsun9/ModuleCache -v foundation-test.swift
  Swift version 5.7.2 (swift-5.7.2-RELEASE)
  Target: x86_64-unknown-linux-gnu
  <unknown>:0: error: cannot load underlying module for 'Dispatch'
```

**Comment by Bo98** [comment]
Excellent, nearly there. Test failure was good as I could see the output bottle and easily identified the issue.

Some of the pain here is me fixing a longstanding issue where the build occasionally failed on Linux due to a race condition. I'm making sure that is fixed here in case we get unlucky during icu4c.

**Review by SMillerDev — DISMISSED** [review]

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
only test failure on `12`, should be ready to go.

**Comment by SMillerDev** [comment]
That's a weird error though:
```
  ==> /usr/local/Cellar/swift/5.7.2/bin/swift build --verbose --disable-sandbox
  warning: /Users/brew/Library/org.swift.swiftpm/configuration is not accessible or not writable, disabling user-level configuration features.
  warning: /Users/brew/Library/org.swift.swiftpm/security is not accessible or not writable, disabling user-level security features.
  warning: /Users/brew/Library/Caches/org.swift.swiftpm is not accessible or not writable, disabling user-level cache features.
  warning: 'swiftpmtest': failed loading cached manifest for 'swiftpmtest': Error Domain=NSCocoaErrorDomain Code=4 "The file “manifests” doesn’t exist." UserInfo={NSFilePath=/Users/brew/Library/Caches/org.swift.swiftpm/manifests, NSUnderlyingError=0x600003848d50 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
  warning: 'swiftpmtest': /usr/local/Cellar/swift/5.7.2/Swift-5.7.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file /private/tmp/swift-test-20230105-98617-17tcqpl/swiftpmtest/Package.swift -target x86_64-apple-macosx10.15.4 -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk -I /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib -I /usr/local/Cellar/swift/5.7.2/Swift-5.7.xctoolchain/usr/lib/swift/pm/ManifestAPI -F /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks -module-cache-path /private/tmp/swift-test-20230105-98617-17tcqpl/ModuleCache -swift-version 5 -package-description-version 5.7.0 -new-driver-path /usr/local/Cellar/swift/5.7.2/Swift-5.7.xctoolchain/usr/bin/swift-driver -disable-implicit-concurrency-module-import -disable-implicit-string-processing-module-import -empty-abi-descriptor -resource-dir /usr/local/opt/swift/Swift-5.7.xctoolchain/usr/lib/swift -module-name main -target-sdk-version 13.0 -o /private/tmp/TemporaryDirectory.NXNVst/Package-1.o
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld /private/tmp/TemporaryDirectory.NXNVst/Package-1.o /usr/local/opt/swift/Swift-5.7.xctoolchain/usr/lib/swift/clang/lib/darwin/libclang_rt.osx.a -F /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks -syslibroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk -lobjc -lSystem -arch x86_64 -force_load /usr/local/opt/swift/Swift-5.7.xctoolchain/usr/lib/swift/macosx/libswiftCompatibilityConcurrency.a -L /usr/local/opt/swift/Swift-5.7.xctoolchain/usr/lib/swift/macosx -L /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk/usr/lib/swift -rpath /usr/lib/swift -platform_version macos 10.15.4 13.0.0 -L /usr/local/Cellar/swift/5.7.2/Swift-5.7.xctoolchain/usr/lib/swift/pm/ManifestAPI -L /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib -lPackageDescription -rpath /usr/local/Cellar/swift/5.7.2/Swift-5.7.xctoolchain/usr/lib/swift/pm/ManifestAPI -o /private/tmp/TemporaryDirectory.UOGJW1/swiftpmtest-manifest
  Swift version 5.7.2 (swift-5.7.2-RELEASE)
  Target: x86_64-apple-macosx10.15.4
  <unknown>:0: remark: did not find a prebuilt standard library for target 'x86_64-apple-macos' compatible with this Swift compiler; building it may take a few minutes, but it should only happen once for this combination of compiler and target
  warning: 'swiftpmtest': failed storing manifest for 'swiftpmtest' in cache: Error Domain=NSCocoaErrorDomain Code=4 "The file “manifests” doesn’t exist." UserInfo={NSFilePath=/Users/brew/Library/Caches/org.swift.swiftpm/manifests, NSUnderlyingError=0x60000384b030 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
  warning: 'swiftpmtest': failed closing cache: Error Domain=NSCocoaErrorDomain Code=4 "The file “manifests” doesn’t exist." UserInfo={NSFilePath=/Users/brew/Library/Caches/org.swift.swiftpm/manifests, NSUnderlyingError=0x60000384b270 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
  Building for debugging...
```

**Comment by Bo98** [comment]
It's a timeout error because of:

> I've disabled the macOS 13 SDK precompile as a result. This means brew test swift will be close to timeout on macOS 13 and maybe macOS 12. 

Let me think of something to do.

**Comment by Bo98** [comment]
Ok forced the test to use macOS 12 SDK for now. To do this I had to fix an issue with SwiftPM where it didn't work when used on a SDK earlier than what it was compiled with.

**Review by cho-m — APPROVED** [review]

**Comment by cho-m** [comment]
LGTM. Added `automerge-skip` in case anyone else wants to comment. @Bo98, you can remove tag whenever you want.

**Review by SMillerDev — APPROVED** [review]

**Review by p-linnane — APPROVED** [review]
Awesome job and thank you for working so hard on this @Bo98!

**Comment by cho-m** [comment]
Removing `automerge-skip` now. Almost ready to continue `icu4c` PR once we finish up #119811

**Comment by devahmedshendy** [comment]
@SMillerDev I found your comment while trying to search for my issue:
```
> brew install chargepoint/xcparse/xcparse

==> Fetching chargepoint/xcparse/xcparse
==> Cloning https://github.com/ChargePoint/xcparse.git
Updating ~/Library/Caches/Homebrew/xcparse--git
==> Checking out tag 2.3.1
HEAD is now at 5ec5931 Merge pull request #76 from ChargePoint/revert-makefile
HEAD is now at 5ec5931 Merge pull request #76 from ChargePoint/revert-makefile
==> Installing xcparse from chargepoint/xcparse
==> make install prefix=/opt/homebrew/Cellar/xcparse/2.3.1
Last 15 lines from ~/Library/Logs/Homebrew/xcparse/01.make:
make
install
prefix=/opt/homebrew/Cellar/xcparse/2.3.1

swift build -c release --disable-sandbox
warning: ~/Library/org.swift.swiftpm/configuration is not accessible or not writable, disabling user-level configuration features.
warning: ~/Library/org.swift.swiftpm/security is not accessible or not writable, disabling user-level security features.
warning: ~/Library/Caches/org.swift.swiftpm is not accessible or not writable, disabling user-level cache features.
warning: 'xcparse-20230312-33748-vqnma8': failed loading cached manifest for 'xcparse-20230312-33748-vqnma8': Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “manifests” in the folder “org.swift.swiftpm”." UserInfo={NSFilePath=~/Library/Caches/org.swift.swiftpm/manifests, NSUnderlyingError=0x6000005fa190 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}
warning: 'xcparse-20230312-33748-vqnma8': failed closing cache: Error Domain=NSCocoaErrorDomain Code=513 "You don’t have permission to save the file “manifests” in the folder “org.swift.swiftpm”." UserInfo={NSFilePath=~/Library/Caches/org.swift.swiftpm/manifests, NSUnderlyingError=0x6000005eb180 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}
error: 'xcparse-20230312-33748-vqnma8': Invalid manifest
/private/tmp/xcparse-20230312-33748-vqnma8/Package.swift:4:8: error: no such module 'PackageDescription'
import PackageDescription
       ^
make: *** [build] Error 1

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/chargepoint/homebrew-xcparse/issues
  ```

What do you suggest, should I make an issue here or `homebrew-xcparse`? or something else I should do .. btw, I reinstalled brew and didn't solve the issue.

**Pull request closed without merging**